### ISSUE TITLE
Missing background color in HTML output

### ISSUE THREAD

**Issue by corryms1993** [pymupdf/PyMuPDF#459]
Hi there,

I've noticed that when running text extraction there are elements appearing at the bottom of the output HTML rather than in their correctly ordered position within the input PDF. 

Example PDF: (https://www.cqc.org.uk/sites/default/files/new_reports/AAAG0004.pdf)

![image](https://user-images.githubusercontent.com/49235569/76099743-78e8b700-5fc3-11ea-9e51-98ec72bb8c80.png)

For example, for the above image, the text: "This section is primarily information for the provider" appears at the top of the PDF. However in the output, the value appears in the following location:

![image](https://user-images.githubusercontent.com/49235569/76100829-46d85480-5fc5-11ea-8e2c-321ab5422f4e.png)

I know that elements within a PDF aren't ordered but I know PyMuPDF handles columns well so thought this seemed strange. Does anyone know if there is a parameter that can be modified to alter the ordering or have any other tips for correcting this? 

Hopefully someone is able to determine the cause of the problem.

Configuration

Windows 10 x64
Python 3.6
PyMuPDF version 1.16.11 (have tried older versions too)

Many thanks,

Corry

**Comment by JorjMcKie** [comment]
(X)HTML and XML output are generated directly by the underlying library MuPDF, so very probably the responsibility for this phenomenon lies there. As written in the PyMuPDF documentation, this output is not perfect.
To confirm please use ``mutool draw -o page-n.html input.pdf n`` with the MuPDF command line tool.
If the error is reproducible with this, then please send a report to their bug site https://bugs.ghostscript.com/enter_bug.cgi.
I will also give it this try.

**Comment by JorjMcKie** [comment]
I just tried like indicated in my previous post. Result:
* line "This section is primarily information for the provider" is invisible because the font is white on white background. When changing text color to something else with a editor, it appears where it should - at least roughly. Just try it out. So the real issue is that the background color is not correctly set. Same is true for text "Requirement notices" (white on white) and others.
* the text "Actionwehavetoldtheprovidertotake" appears like this - w/o intervening spaces. This is a duplicate of #456.

With these modifications, the html page really looks like this

![grafik](https://user-images.githubusercontent.com/8290722/76106655-ae82a580-5fad-11ea-8901-0356d62cc5f0.png)

So, bottom line there are two issues:

1. incorrect suppression of spaces (a duplicate issue #456 ).
2. missing background color.

And a minor issue of providing correct bounds reflecting the html page width.

**Comment by JorjMcKie** [comment]
Depending on your motivation for producing HTML output, here is something which might help:
The SVG output of this PDF's pages looks exactly like the original. SVG files can be used in a number of contexts, and be displyed by internet prowsers:

```python
doc = fitz.open("AAAG0004.pdf")
page = doc[-1]  # last page
svgfile = open("page-%i.svg" % page.number, "w")
svgtext = page.getSVGimage()
svgfile.write(svgtext)
svgfile.close()
```

**Comment by JorjMcKie** [comment]
Assuming we have reached a satisfactory discussion status, I am closing the issue.
